osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmap.cs

```diff
@@ -134,6 +157,14 @@ private void load(SongSelect songSelect, BeatmapManager manager, BeatmapSetOverl
             };
         }
 
+        private ScoreInfo getTopScore(ScoreManager scoreManager, int beatmapID, int rulesetID)
+        {
+            return scoreManager
+                .QueryScores(s => !s.DeletePending && s.Beatmap.ID == beatmapID && s.Ruleset.ID == rulesetID)
+                .OrderByDescending(s => s.TotalScore)
+                .FirstOrDefault();
+        }
```

Comment:
Have you checked the performance implications of this? sqlite is in-memory, so it shouldn't be horrible, but I'm a bit concerned this will tank performance for tens of thousands of beatmaps even more than what we already have (and yes, people are actually importing that many). May be wise to go with some sort of caching component that will do a `GroupBy` and compute all of this in one shot instead of querying one by one, but that needs benchmarking.

Not necessarily an immediate concern but I imagine it's a potential issue.

Reply:
Yeah this is one of those cases that should probably only be loaded when the drawable is displayed.

Furthermore, this shows e.g. SS ranks from osu! mode even if the user has osu!mania mode selected in the toolbar. Splitting this out into its own component that loads in a lazy manner will also help with responding to the currently-loaded ruleset.